### PR TITLE
Data catalog tests

### LINKED ISSUE #937: Enable integration tests in `tests/test_data_catalog.py`
### Kind of request

None

### Enhancement Description

enable integration tests marked as skip or commented out in `tests/test_data_catalog.py`or add a further explanation why they can't be. 

### Use case

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by savente93**
## Issue addressed
Parially Fixes #937 

## Explanation
To avoid giant mega PRs I'll also submit at least a partial PR. This one at least migrates the deltares data to the new format as well. 

## General Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `v1`
- [ ] Tests & pre-commit hooks pass
- [ ] Updated documentation
- [ ] Updated changelog.rst

### CODE REVIEW

**Review by Jaapel — CHANGES_REQUESTED** [review]
Some things are not correct. Perhaps nice to merge some of the gcsfs and aws catalog changes in test_data_adapter.py here to test those.

**Comment by savente93** [comment]
Thanks! I'll get that fixed asap

File: data/catalogs/artifact_data/registry.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-v0.0.6/data_catalog.yml 5d9e47158185f1afbf793db68c887f6e6b119d7ffd3edfbc198e5ae3a9d760f3
-v0.0.7/data_catalog.yml 8c4aa8e5bc28fd9a6d25b93d73dc091dd9aa6beec3de48dc2a56b57aafe415ee
 v0.0.8/data_catalog.yml c1cf2229eeb93607ea881fdbced45ebf43471271f0b7d878db8183734274cd88
+v0.0.7/data_catalog.yml 8c4aa8e5bc28fd9a6d25b93d73dc091dd9aa6beec3de48dc2a56b57aafe415ee
 v0.0.9/data_catalog.yml a074379f3ef244f3a860ec40165163538b6d690d8a3cbc8c6e883e02a8936258
+v0.0.6/data_catalog.yml 5d9e47158185f1afbf793db68c887f6e6b119d7ffd3edfbc198e5ae3a9d760f3
 v1.0.0/data_catalog.yml c8600a213e1baf397002ec0d2e2f825ae938c2a1ab266d84011356413f4d5617
```
[the remark this answers is not in the document]
Comment (savente93) [reply data/catalogs/artifact_data/registry.txt:5]: by accident I think I'll revert them

File: data/catalogs/gcs_cmip6_data/v1.0.0/data_catalog.yml [hunk 2]
```diff
@@ -0,0 +1,370 @@
+---
```
[the remark this answers is not in the document]
Comment (savente93) [reply data/catalogs/gcs_cmip6_data/v1.0.0/data_catalog.yml]: should be fixed

File: data/catalogs/deltares_data/v0.7.0/data_catalog.yml [hunk 3]
```diff
@@ -0,0 +1,1881 @@
...
+      driver: raster
+      kwargs:
+        chunks:
+          x: 150
+          y: 360
+      path: hydro/glofas_era5/glofas_v31_uparea.tif
+      rename:
+        glofas_v31_uparea: uparea
+      unit_mult:
+        uparea: 0.000001
```
Comment (Jaapel) [inline data/catalogs/deltares_data/v0.7.0/data_catalog.yml:884]: @DirkEilander I migrated an alias here to a new variant, or do we want these to be different entries?

File: hydromt/data_catalog.py [hunk 4]
```diff
@@ -424,14 +426,14 @@ def add_source(self, name: str, source: DataSource) -> None:
...
+            version = str(source.version)
         else:
             version = "_UNSPECIFIED_"  # make sure this comes first in sorted list
 
         if source.provider:
-            provider = source.provider
+            provider = str(source.provider)
         else:
-            provider = "_UNSPECIFIED_"
+            provider = source.driver.filesystem.protocol[0]
```
Comment (Jaapel) [inline hydromt/data_catalog.py:436]: Protocol can be
```python
pr: Union[str, Tuple[str, ...]] = _fs.protocol
```
```suggestion
            protocol = source.driver.filesystem.protocol
            if isinstance(protocol, str):
                provider: str = protocol
            else:
                provider: str = protocol[0]
```

**Review by Jaapel — APPROVED** [review]
Let's just do it. One small thing though

**Comment by savente93** [comment]
I'll continue working on the skipped tests from next week on, so it should all be fixed by the next release.

**Pull request merged**